data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoClientException.java

```diff
@@ -0,0 +1,16 @@
+package com.microsoft.azure.kusto.data.exceptions;
+
+public abstract class KustoClientException extends Exception {
+    public KustoClientException(String message) {
+        super(message);
+    }
+
+    public KustoClientException(String message, Exception exception) {
+        super(message, exception);
+    }
+
+    /*
+      Can return null if permanency is not known
```

Comment:
Not a big fan of a Boolean that is also a tristate,
maybe this should be an enum/constants